### ISSUE TITLE
t3000-mpi-basic.t: failing in CI tests

### ISSUE THREAD

**Issue by SteVwonder** [flux-framework/flux-core#3463]
For both (https://github.com/flux-framework/flux-core/pull/3459) and (https://github.com/flux-framework/flux-core/pull/3461), the MPI basic tests failed.  Not sure why.  The changes in those two tests are pretty different (and one should effectively be a no-op in terms of the CI), so I wonder if it is a new intermittent failure:

```
2021-01-08T19:03:15.2632900Z ##[warning]Found 1 errors from 259 tests in testsuite
2021-01-08T19:03:15.2645402Z ##[error]not ok 2 - mpi hello all ranks
2021-01-08T19:03:15.2647424Z ##[error]ERROR: t3000-mpi-basic.t - exited with status 1
2021-01-08T19:03:15.2649167Z ##[group]./flux-core-0.22.0-76-gcec8c98b0/_build/sub/t/t3000-mpi-basic.output
2021-01-08T19:03:15.2649856Z expecting success: 
2021-01-08T19:03:15.2650363Z 	run_program 5 1 1 ${FLUX_BUILD_DIR}/t/mpi/hello >single.$OPTS
2021-01-08T19:03:15.2650715Z 
2021-01-08T19:03:15.2651167Z ok 1 - mpi hello singleton
2021-01-08T19:03:15.2651551Z mpi hello singleton
2021-01-08T19:03:15.2651803Z 
2021-01-08T19:03:15.2652130Z expecting success: 
2021-01-08T19:03:15.2652601Z 	run_program 5 ${SIZE} ${SIZE} ${FLUX_BUILD_DIR}/t/mpi/hello \
2021-01-08T19:03:15.2653059Z 		| tee allranks.$OPTS \
2021-01-08T19:03:15.2653674Z 		&& grep -q "There are ${SIZE} tasks" allranks.$OPTS
2021-01-08T19:03:15.2654003Z 
2021-01-08T19:03:15.2654440Z not ok 2 - mpi hello all ranks
2021-01-08T19:03:15.2654696Z 
2021-01-08T19:03:15.2655008Z # failed 1 among 2 test(s)
```

**Comment by grondo** [comment]
I was able to reproduce this on the GH action running VM.

The problem appears to be that once in every several runs, the "all ranks" MPI_Init is taking >5s

```
0: completed MPI_Init in 5.073s.  There are 4 tasks
0: completed first barrier in 0.036s
0: completed MPI_Finalize in 0.033s
```

Thus the job is timed out and the test fails.

I don't know the cause yet of the 5s delay. For now, our testsuite would likely be more reliable if we increased the timeout in these tests. Meanwhile, I'll see if the problem is reproducible outside of GH actions for easier debugging.
